packages/grid/x-data-grid/src/hooks/features/dimensions/useGridDimensions.ts

```diff
@@ -79,6 +78,7 @@ export function useGridDimensions(
     const rootElement = apiRef.current.rootElementRef?.current;
     const columnsTotalWidth = gridColumnsTotalWidthSelector(apiRef);
     const pinnedRowsHeight = calculatePinnedRowsHeight(apiRef);
+    const rowsMeta = gridRowsMetaSelector(apiRef.current.state);
```

Comment:
The `rowsMeta.currentPageTotalHeight` value was outdated here, so I moved `rowsMeta` inside the callback.
For example, when filtering the rows and there were no rows matching the filtering criteria, `updateGridDimensionsRef` was called with the outdated value of `rowsMeta.currentPageTotalHeight` before applying filters.